sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp

```diff
@@ -448,3 +455,1201 @@ ZER_APIEXPORT zer_result_t ZER_APICALL zerPlatformGetInfo(
+    return ReturnValue(uint32_t{1});
+  case ZER_DEVICE_INFO_LINKER_AVAILABLE:
+    return ReturnValue(uint32_t{1});
+  case ZER_DEVICE_INFO_MAX_COMPUTE_UNITS: {
+    uint32_t MaxComputeUnits =
+        Device->ZeDeviceProperties->numEUsPerSubslice *
+        Device->ZeDeviceProperties->numSubslicesPerSlice *
+        Device->ZeDeviceProperties->numSlices;
+
+    bool RepresentsCSlice =
+        Device->QueueGroup[_zer_device_handle_t::queue_group_info_t::Compute]
+            .ZeIndex >= 0;
+    if (RepresentsCSlice)
+      MaxComputeUnits /= Device->RootDevice->SubDevices.size();
+
+    return ReturnValue(uint32_t{MaxComputeUnits});
+  }
+  case ZER_DEVICE_INFO_MAX_WORK_ITEM_DIMENSIONS:
+    // Level Zero spec defines only three dimensions
+    return ReturnValue(uint32_t{3});
+  case ZER_DEVICE_INFO_MAX_WORK_GROUP_SIZE:
+    return ReturnValue(
+        uint64_t{Device->ZeDeviceComputeProperties->maxTotalGroupSize});
+  case ZER_DEVICE_INFO_MAX_WORK_ITEM_SIZES: {
+    struct {
+      size_t Arr[3];
+    } MaxGroupSize = {{Device->ZeDeviceComputeProperties->maxGroupSizeX,
+                       Device->ZeDeviceComputeProperties->maxGroupSizeY,
+                       Device->ZeDeviceComputeProperties->maxGroupSizeZ}};
+    return ReturnValue(MaxGroupSize);
+  }
+  case ZER_EXT_DEVICE_INFO_MAX_WORK_GROUPS_3D: {
+    struct {
+      size_t Arr[3];
+    } MaxGroupCounts = {{Device->ZeDeviceComputeProperties->maxGroupCountX,
+                         Device->ZeDeviceComputeProperties->maxGroupCountY,
+                         Device->ZeDeviceComputeProperties->maxGroupCountZ}};
+    return ReturnValue(MaxGroupCounts);
+  }
+  case ZER_DEVICE_INFO_MAX_CLOCK_FREQUENCY:
+    return ReturnValue(uint32_t{Device->ZeDeviceProperties->coreClockRate});
+  case ZER_DEVICE_INFO_ADDRESS_BITS: {
+    // TODO: To confirm with spec.
+    return ReturnValue(uint32_t{64});
+  }
+  case ZER_DEVICE_INFO_MAX_MEM_ALLOC_SIZE:
+    return ReturnValue(uint64_t{Device->ZeDeviceProperties->maxMemAllocSize});
+  case ZER_DEVICE_INFO_GLOBAL_MEM_SIZE: {
+    uint64_t GlobalMemSize = 0;
+    for (const auto &ZeDeviceMemoryExtProperty :
+         Device->ZeDeviceMemoryProperties->second) {
+      GlobalMemSize += ZeDeviceMemoryExtProperty.physicalSize;
+    }
+    return ReturnValue(uint64_t{GlobalMemSize});
+  }
+  case ZER_DEVICE_INFO_LOCAL_MEM_SIZE:
+    return ReturnValue(
+        uint64_t{Device->ZeDeviceComputeProperties->maxSharedLocalMemory});
+  case ZER_DEVICE_INFO_IMAGE_SUPPORTED:
+    return ReturnValue(
+        uint32_t{Device->ZeDeviceImageProperties->maxImageDims1D > 0});
+  case ZER_DEVICE_INFO_HOST_UNIFIED_MEMORY:
+    return ReturnValue(uint32_t{(Device->ZeDeviceProperties->flags &
+                                 ZE_DEVICE_PROPERTY_FLAG_INTEGRATED) != 0});
+  case ZER_DEVICE_INFO_AVAILABLE:
+    return ReturnValue(uint32_t{ZeDevice ? true : false});
+  case ZER_DEVICE_INFO_VENDOR:
+    // TODO: Level-Zero does not return vendor's name at the moment

```

Comment:
L0 already returning this. we can follow up on this in a follow-up patch.

Reply:
OK